blocks/algorithms/__init__.py

```diff
@@ -590,3 +590,29 @@ def compute_step(self, param, previous_step):
                    (time, t1)]
 
         return step, updates
+
+
+class RemoveNotFinite(StepRule):
+    """A step rule that replaces non-finite elements.
+
+    Replaces non-finite elements (`inf` or `NaN`) in the step with
```

Comment:
Is this really what it does? If I read the code it seems like it replaces the *entire* step with the scaled parameter, instead of doing it entry-wise (which is how I read this docstring).